### PR TITLE
Feature: responsive sidebar

### LINKED ISSUE #158: Fix responsiveness issues in GE V2
Repro Steps: 
1. Go to http://graphtryit-staging.azurewebsites.net/ on Chrome
2. Right click the page and select inspect
3. Select Toggle Device Toolbar
4. Switch between screen sizes. the UI is distorted. Below is a screenshot.

image

I scrolled the same page and took the below screenshot
image

### PR DESCRIPTION
**Pull request by cwahome**
## Overview

- Ensures the sidebar is responsive on multiple devices
Fixes #158 

Auxilliary goals
- Adds a horizontal sidebar to monaco editor

### Demo

Sidebar hidden
image

Sidebar visible

image

### Notes

The sidebar is hidden by default on small screens. the hamburger menu toggles its visibility

## Testing Instructions

# without dev tools
* Load the app on your phone using the ip url provided on the terminal

# with dev tools
* Load the app on your browser
* Open dev tools and switch between the different screen sizes provided

### CODE REVIEW

File: src/app/views/App.tsx [hunk 1]
```diff
@@ -165,64 +172,86 @@ class App extends Component<IAppProps, IAppState> {
...
-                      </a>
-                    </p>
+                    {`${error.statusText} - ${error.status}`}
                   </MessageBar>
-                </div>
-              )}
-              <div style={{ marginBottom: 8 }}>
-                <QueryRunner onSelectVerb={this.handleSelectVerb} />
+                )}
+                {sampleQuery && (<div />)}
```
Comment (Kachulio1) [inline src/app/views/App.tsx]: I think it's safe to remove this line, I don't see `sampleQuery` passed as a prop if you look at `mapDispatchToProps ||  mapStateToProps` but there is `receivedSampleQuery` not sure if that's what should be there.

File: src/app/views/App.tsx [hunk 2]
```diff
@@ -165,64 +172,86 @@ class App extends Component<IAppProps, IAppState> {
...
+                    title='Remove sidebar'
+                    ariaLabel='Remove sidebar'
+                    onClick={() => this.toggleSidebar()}
+                  />}
+                  {showSidebar && <Sidebar />}
+                </div>
+              )}
+              <div className={layout}>
+                {graphExplorerMode === Mode.Complete && <Authentication />}
+                {graphExplorerMode === Mode.TryIt && (
```
Comment (Kachulio1) [inline src/app/views/App.tsx]: `graphExplorerMode` can either be `Complete or TryIt ` we don't need multiple checks. we can just use the complete check and return Authentication or the message bar.

File: src/app/views/App.tsx [hunk 3]
```diff
@@ -41,17 +40,25 @@ interface IAppProps {
...
     this.state = {
-      selectedVerb: 'GET'
+      selectedVerb: 'GET',
+      showSidebar: true,
+      showToggle: false,
     };
   }
 
   public componentDidMount = () => {
+    const mediaQueryList = window.matchMedia('(max-width: 767px)');
```
Comment (Kachulio1) [inline src/app/views/App.tsx]: This is nice 👍

File: src/app/views/App.tsx [hunk 4]
```diff
@@ -41,17 +40,25 @@ interface IAppProps {
...
+    this.displayToggleButton(mediaQueryList);
+    mediaQueryList.addListener(this.displayToggleButton);
```
Comment (jobala) [inline src/app/views/App.tsx]: Make sure to remove the listener in `componentWillUnMount`. Not removing listeners is a common cause for memory leaks.

File: src/app/views/App.tsx [hunk 5]
```diff
@@ -165,64 +172,86 @@ class App extends Component<IAppProps, IAppState> {
...
+
   public render() {
     const classes = classNames(this.props);
     const { graphExplorerMode, error, actions, sampleQuery }: any = this.props;
+    const { showSidebar, showToggle } = this.state;
     const layout =
       graphExplorerMode === Mode.TryIt
-        ? 'col-sm-12'
-        : 'col-sm-12 col-lg-9';
+        ? 'col-sm-12 col-xs-12'
```
Comment (jobala) [inline src/app/views/App.tsx]: `col-xs-12` will suffice.

The column size for the smallest breakpoint is used for other breakpoints when they are not specified.

File: src/app/views/App.tsx [hunk 6]
```diff
@@ -165,64 +172,86 @@ class App extends Component<IAppProps, IAppState> {
...
+        : 'col-xs-12 col-sm-12 col-lg-9 col-md-8';
```
Comment (jobala) [inline src/app/views/App.tsx]: What is the reason behind using 8 columns for medium screen sizes?

Comment (cwahome) [reply src/app/views/App.tsx]: I noticed that for tablets to render properly, I'd need to use the col-md-8 and col-md-4 combination

**Comment by cwahome** [comment]
> When the hamburger icon is toggled on the Request and Response components are stacked beneath it, meaning users have to scroll down to see the result of the response. What do you think of hiding the Sidebar after clicking on a sample?|
> 

Interesting idea. I can look at it.


> Also the authentication component is broken for logged in users.

I believe I fixed it in the work i was doing on the profile component in PR: #176

File: src/app/views/App.tsx (hunk 1 above)
Comment (jobala) [reply src/app/views/App.tsx]: @cwahome you can remove this now.

**Comment by andrueastman** [comment]
LGTM

**Pull request merged**